actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -30,6 +30,30 @@ module Renderers
     included do
       class_attribute :_renderers, default: Set.new.freeze
       class_attribute :escape_json_responses, instance_accessor: false, default: true
+
+      redefine_method :escape_json_responses= do |value|
+        if value
+          ActionController.deprecator.warn(<<~MSG.squish)
+            Setting action_controller.escape_json_responses = true is deprecated and will have no effect in Rails 8.2.
+            Set it to `false` or use `config.load_defaults(8.1)`.
```

Comment:
wdyt about this? https://github.com/rails/rails/pull/54643#discussion_r2022206660

```suggestion
            Set it to `false`, or remove the config.
```